app/controllers/users_controller.rb

```diff
@@ -19,46 +27,80 @@ def new
 
 private
 
-  def invite_user
+  def setup_user_in_cognito
     temporary_password = ('a'..'z').to_a.sample(3) + ('A'..'Z').to_a.sample(3) + ('0'..'9').to_a.sample(3) + ('!'..'/').to_a.sample(1)
-    begin
-      invite = SelfService.service(:cognito_client).admin_create_user(
-        temporary_password: temporary_password.join(''),
-        user_attributes: [
-          {
-            name: 'email',
-            value: @form.email
-          },
-          {
-            name: 'given_name',
-            value: @form.given_name
-          },
-          {
-            name: 'family_name',
-            value: @form.family_name
-          },
-          {
-            name: 'custom:roles',
-            value: @form.roles.join(",")
-          }
-        ],
-        username: @form.email,
-        user_pool_id: Rails.application.secrets.cognito_user_pool_id
+    SelfService.service(:cognito_client).admin_create_user(
+      temporary_password: temporary_password.join(''),
+      user_attributes: [
+        {
+          name: 'email',
+          value: @form.email
+        },
+        {
+          name: 'given_name',
+          value: @form.given_name
+        },
+        {
+          name: 'family_name',
+          value: @form.family_name
+        },
+        {
+          name: 'custom:roles',
+          value: @form.roles.join(",")
+        }
+      ],
+      username: @form.email,
+      user_pool_id: user_pool_id
       )
+  end
+
+  def add_user_to_team_in_cognito(new_user, team)
+    SelfService.service(:cognito_client).admin_add_user_to_group(
+      user_pool_id: user_pool_id,
+      username: new_user.username,
+      group_name: team.name
+    )
+  end
+
+  def invite_user
+    begin
+      if params[:team_id] != '0'
+        raise StandardError(t('users.invite.errors.team_missing')) unless Team.exists?(params[:team_id])
+
+        team = Team.find(params[:team_id])
+      else
+        team = Team.new(id: 0, name: 'gds')
+      end
+      invite = setup_user_in_cognito
     rescue Aws::CognitoIdentityProvider::Errors::AliasExistsException,
            Aws::CognitoIdentityProvider::Errors::UsernameExistsException => e
       flash.now[:errors] = t('users.invite.errors.already_exists')
     rescue StandardError => e
       flash.now[:errors] = t('users.invite.errors.generic_error')
     end
 
+    begin
+      add_user_to_team_in_cognito(invite.user, team) unless team.nil?
+    rescue StandardError => e
```

Comment:
We should be more specific about the rescue. It will be Cognito based error. There is is a catch all I think.